proxstar/vnc.py

```diff
@@ -31,38 +36,80 @@ def get_vnc_targets():
-    target = next((target for target in targets if target['port'] == port), None)
+    target = next((target for target in targets if target['host'] == f'{node}:{port}'), None)
     if target:
+        print('Host is already in the targets file')
         return target['token']
     else:
         target_file = open(app.config['WEBSOCKIFY_TARGET_FILE'], 'a')
         token = gen_password(32, 'abcdefghijklmnopqrstuvwxyz0123456789')
-        target_file.write('{}: 127.0.0.1:{}\n'.format(token, str(port)))
+        target_file.write(f'{token}: {node}:{port}\n')
         target_file.close()
         return token
 
 
-def delete_vnc_target(port):
+def delete_vnc_target(node, port):
     targets = get_vnc_targets()
-    target = next((target for target in targets if target['port'] == str(port)), None)
+    target = next((target for target in targets if target['host'] == f'{node}:{port}'), None)
     if target:
         targets.remove(target)
         target_file = open(app.config['WEBSOCKIFY_TARGET_FILE'], 'w')
         for target in targets:
-            target_file.write('{}: 127.0.0.1:{}\n'.format(target['token'], target['port']))
+            target_file.write(f"{target['token']}: {target['host']}\n")
         target_file.close()
 
 
+def open_vnc_session(vmid, node, proxmox_user, proxmox_pass):
+    """Pings the Proxmox API to request a VNC Proxy connection. Authenticates
+    against the API using a Uname/Pass, gets a few tokens back, then uses those
+    tokens to  open the VNC Proxy. Use these to connect to the VM's host with
+    websockify proxy.
+    Returns: Ticket to use as the noVNC password, and a port.
+    """
+    # Get Proxmox API ticket and CSRF_Prevention_Token
+    # TODO (willnilges): Use Proxmoxer to get this information
+    # TODO (willnilges): Report errors
+    data = {'username': proxmox_user, 'password': proxmox_pass}
+    response_data = requests.post(
+        f'https://{node}.csh.rit.edu:8006/' + 'api2/json/access/ticket',
+        verify=False,
+        data=data,
+    ).json()['data']
+    if response_data is None:
+        raise requests.AuthenticationError(
+            'Could not authenticate against `ticket` endpoint! Check uname/password'
+        )
+    csrf_prevention_token = response_data['CSRFPreventionToken']
+    ticket = response_data['ticket']
+    proxy_params = {'node': node, 'vmid': str(vmid), 'websocket': '1', 'generate-password': '0'}
+    vncproxy_response_data = requests.post(
+        f'https://{node}.csh.rit.edu:8006/api2/json/nodes/{node}/qemu/{vmid}/vncproxy',
+        verify=False,
+        timeout=5,
+        params=proxy_params,
+        headers={'CSRFPreventionToken': csrf_prevention_token},
+        cookies={'PVEAuthCookie': ticket},
+    ).json()['data']
+
+    return urllib.parse.quote_plus(vncproxy_response_data['ticket']), vncproxy_response_data['port']
+
+
 def start_ssh_tunnel(node, port):
+    """Forwards a port on a node
+    to the proxstar container
+    """
```

Comment:
Maybe comment here that this is deprecated? I believe there is a deprecated tag you can maybe add since these are no longer in use (I think).

Reply:
Where the fuck is this comment?

Reply:
Added tag